packages/core/src/amazonqFeatureDev/controllers/chat/controller.ts

```diff
@@ -300,6 +306,29 @@ export class FeatureDevController {
         this.messenger.sendAsyncEventProgress(tabID, false, undefined)
     }
 
+    private printFailedCodeGenMessage(session: Session, message: string, tabID: string, retryable: boolean) {
```

Comment:
question: isn't this function missing the chat prompt blocking?

Reply:
The prompt is blocked on line 397. It's the last step of the function that calls this one.